packages/wrangler/e2e/types.test.ts

```diff
@@ -29,159 +29,110 @@ const seed = {
 		const helper = new WranglerE2ETestHelper();
 		await helper.seed(seed);
 		const output = await helper.run(`wrangler types`);
 
-		expect(output.stdout).not.toContain(`Generating runtime types...`);
+		expect(output.stdout).toContain("Generating runtime types...");
+		expect(output.stdout).toContain("Runtime types generated.");
+		expect(output.stdout).toContain(
+			"✨ Types written to worker-configuration.d.ts"
+		);
+		expect(output.stdout).toContain("📖 Read about runtime types");
 	});
 
-	it("should generate runtime types at the default path", async () => {
+	it("should generate runtime types and env types in one file at the default path", async () => {
 		const helper = new WranglerE2ETestHelper();
 		await helper.seed(seed);
-		const output = await helper.run(`wrangler types --x-include-runtime`);
-
-		const fileExists = existsSync(
-			path.join(helper.tmpPath, "./.wrangler/types/runtime.d.ts")
-		);
-
-		expect(fileExists).toEqual(true);
-		expect(output.stdout).toContain(`Generating runtime types...`);
-		expect(output.stdout).toContain(`Generating project types...`);
-		expect(output.stdout).toContain(
-			`✨ Runtime types written to ./.wrangler/types/runtime.d.ts`
-		);
-		expect(output.stdout).toContain(
-			`"types": ["./.wrangler/types/runtime.d.ts"]`
-		);
+		const output = await helper.run(`wrangler types`);
+		expect(output.stdout).toContain("Generating project types...");
+		expect(output.stdout).toContain("interface Env {");
+		expect(output.stdout).toContain("Generating runtime types...");
+		expect(output.stdout).toContain("Runtime types generated.");
 		expect(output.stdout).toContain(
-			`📣 Since you have Node.js compatibility mode enabled, you should consider adding Node.js for TypeScript by running "npm i --save-dev @types/node@20.8.3". Please see the docs for more details: https://developers.cloudflare.com/workers/languages/typescript/#transitive-loading-of-typesnode-overrides-cloudflareworkers-types`
+			"✨ Types written to worker-configuration.d.ts"
 		);
-		expect(output.stdout).toContain(
-			`Remember to run 'wrangler types --x-include-runtime' again if you change 'compatibility_date' or 'compatibility_flags' in your wrangler.toml file.`
+		const file = readFileSync(
+			path.join(helper.tmpPath, "./worker-configuration.d.ts"),
+			"utf8"
 		);
+		expect(file).contains('declare module "cloudflare:workers"');
+		expect(file).contains("interface Env");
 	});
 
-	it("should generate runtime types at the provided path", async () => {
+	it("should include header with version information in the generated types", async () => {
 		const helper = new WranglerE2ETestHelper();
 		await helper.seed(seed);
-		const output = await helper.run(
-			`wrangler types --x-include-runtime="./types.d.ts"`
-		);
+		await helper.run(`wrangler types "./types.d.ts" `);
 
-		const fileExists = existsSync(path.join(helper.tmpPath, "./types.d.ts"));
+		const file = (await readFile(path.join(helper.tmpPath, "./types.d.ts")))
+			.toString()
```

Comment:
Looks like this can be updated at multiple places
```suggestion
		const file = (await readFile(path.join(helper.tmpPath, "./types.d.ts"), "utf8"))
```